app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -508,11 +508,20 @@ def run_operation(parameters, operation_name = nil)
                                          MiddlewareServer)
           end
         end
-        operation_connection = connection.operations(true)
-        if operation_name.nil?
-          operation_connection.invoke_generic_operation(parameters, &callback)
-        else
-          operation_connection.invoke_specific_operation(parameters, operation_name, &callback)
+
+        begin
+          operation_connection = connection.operations(true)
+
+          if operation_name.nil?
+            operation_connection.invoke_generic_operation(parameters, &callback)
+          else
+            operation_connection.invoke_specific_operation(parameters, operation_name, &callback)
+          end
+        rescue Errno::ECONNREFUSED => e
```

Comment:
perhaps the rescue should be expanded to additional errors. not only connection refused. if hawkular server is up and returns 500 err, for example, imo it can behave the same on miq side